src/coinbase/address/wallet_address.ts

```diff
@@ -285,7 +290,27 @@ export class WalletAddress extends Address {
       throw new Error("Cannot invoke contract from address without private key loaded");
     }
 
-    const contractInvocation = await this.createContractInvocation(options);
+    let atomicAmount: string | undefined;
+
+    if (options.assetId && options.amount) {
+      const asset = await Asset.fetch(this.getNetworkId(), options.assetId);
+      const normalizedAmount = new Decimal(options.amount.toString());
+      const currentBalance = await this.getBalance(options.assetId);
```

Comment:
messaged you about this as well, but to double check my understanding - is it that `normalizedAmount` can be in either `wei`, `gwei`, `ether` and `currentBalance` will fetch the balance in the same unit? this way the comparison works?

and later when we do `toAtomicAmount`, this will convert to `wei`